File: tests/adapter-fixtures/rust/basic.rs
// Fixture: basic Rust entities
// Expected: 2 functions, 1 struct, 1 impl block

pub struct UserProfile {
    pub name: String,
    pub email: String,
}

impl UserProfile {
    pub fn new(name: String, email: String) -> Self {
        Self { name, email }
    }
}

pub fn create_user(name: &str, email: &str) -> UserProfile {
    UserProfile::new(name.to_string(), email.to_string())
}

fn helper_internal() -> u32 {
    42
}

File: tests/adapter-fixtures/cpp/basic.cpp
// Fixture: basic C++ entities
// Expected: 1 class, 2 functions

class UserService {
public:
    UserService() = default;

    int count() const {
        return 1;
    }
};

static int helper_internal() {
    return 42;
}

int create_user() {
    UserService service;
    return service.count() + helper_internal();
}
